cql3/statements/alter_keyspace_statement.cc

```diff
@@ -99,8 +99,8 @@ void cql3::statements::alter_keyspace_statement::validate(query_processor& qp, c
                         // first we need to report non-existing DCs, then if RFs aren't changed by too much.
                         continue;
                     }
-                    if (!validate_rf_difference(old_rf, new_rf)) {
-                        throw exceptions::invalid_request_exception("Cannot modify replication factor of any DC by more than 1 at a time.");
+                    if (total_abs_rfs_diff += get_abs_rf_diff(old_rf, new_rf); total_abs_rfs_diff >= 2) {
```

Comment:
So IMO the regression test you add at the beginning should already contain the full test (with multiple DCs being changed), and then the fix commit should already contain the complete fix. (And any necessary refactors come to a separate commit before that.)

Reply:
> I think it's worth restructuring this PR so it starts with the complete approach right from the beginning.

I crafted smaller commits to present steps necessary to fix every corner case, and that's because I was afraid having most of it in a single commit would make this commit too big and I'd be asked to split it. Let me know if this is really needed. 

> It's a bit confusing that the total_abs_rfs_diff >= 2 check is done in every iteration of the for loop.
> Looks like you want to calculate the sum total_abs_rfs_diff first, and at the end of the loop, validate that the difference hasn't gotten too big.

The check has to be performed in every loop iteration, because if RF changes by +1 in one DC and by -1 in the other, the total difference is equal to 0, but we'd end up having 2 pending replicas.

Reply:
@kbr-scylla so how do we proceed here?

Reply:
> The check has to be performed in every loop iteration, because if RF changes by +1 in one DC and by -1 in the other, the total difference is equal to 0, but we'd end up having 2 pending replicas.

The total difference of absolutes is 2, which is what you get at the end of the iteration

Reply:
> > The check has to be performed in every loop iteration, because if RF changes by +1 in one DC and by -1 in the other, the total difference is equal to 0, but we'd end up having 2 pending replicas.
> 
> The total difference of absolutes is 2, which is what you get at the end of the iteration

As discussed, you're correct. I misunderstood your original question. For posterity: we agreed to leave the check/if in every loop iteration to exit the loop early (at the first possibility), but this check could as well be done outside of the loop, as Kamil suggested.